src/platforms/stm32/src/main.c

```diff
@@ -170,10 +201,31 @@ int main()
 
     AVM_LOGI(TAG, "Starting: %s...\n", startup_module_name);
     context_execute_loop(ctx, mod, "start", 0);
-    AVM_LOGI(TAG, "Return value: %lx", (long) term_to_int32(ctx->x[0]));
 
-    while (1)
-        ;
+    term ret_value = ctx->x[0];
+    char *ret_atom_string = interop_atom_to_string(ctx, ret_value);
+    if (ret_atom_string != NULL) {
+        AVM_LOGI(TAG, "Exited with return: %s", ret_atom_string);
+    } else {
+        AVM_LOGI(TAG, "Exited with return value: %lx", (long) term_to_int32(ret_value));
+    }
 
+    bool reboot_on_not_ok =
+#if defined(CONFIG_REBOOT_ON_NOT_OK)
+        CONFIG_REBOOT_ON_NOT_OK ? true : false;
+#else
+        false;
+#endif
+    if (reboot_on_not_ok && ret_value != OK_ATOM) {
+        AVM_LOGE(TAG, "AtomVM application terminated with non-ok return value.  Rebooting ...");
+        scb_reset_system();
+    } else {
+        AVM_LOGI(TAG, "AtomVM application terminated.  Going to sleep forever ...");
+        // Disable all interrupts
+        cm_disable_interrupts();
+        while (1) {
+            ;
```

Comment:
any sleep functionality, or will this just spin?

Reply:
This will spin for now, so the device can be debugged with gdb. I plan on adding logic later to determine if it the device is connected by jtag so it will spin in that case or put the system into deep sleep mode if it is not attached.